Tests/SwiftGenTests/ConfigTests.swift

```diff
@@ -0,0 +1,267 @@
+      if case TemplateRef.name("swift3") = xcassetsEntries[0].template { /* OK */ } else {
+        XCTFail("Unexpected template: \(xcassetsEntries[0].template)")
+      }
+      XCTAssertEqual(xcassetsEntries[0].output, "assets-colors.swift")
+      XCTAssertEqual(xcassetsEntries[0].parameters.count, 0)
+      // > xcassets[1]
+      XCTAssertEqual(xcassetsEntries[1].paths, ["XCAssets/Images.xcassets"])
+      if case TemplateRef.name("swift3") = xcassetsEntries[1].template { /* OK */ } else {
+        XCTFail("Unexpected template: \(xcassetsEntries[1].template)")
+      }
+      XCTAssertEqual(xcassetsEntries[1].output, "assets-images.swift")
+      XCTAssertEqual(xcassetsEntries[1].parameters.count, 1)
+      XCTAssertEqual(xcassetsEntries[1].parameters["enumName"] as? String, "Pics")
+      // > xcassets[2]
+      XCTAssertEqual(xcassetsEntries[2].paths, ["XCAssets/Colors.xcassets", "XCAssets/Images.xcassets"])
+      if case TemplateRef.name("swift4") = xcassetsEntries[2].template { /* OK */ } else {
+        XCTFail("Unexpected template: \(xcassetsEntries[2].template)")
+      }
+      XCTAssertEqual(xcassetsEntries[2].output, "assets-all.swift")
+      XCTAssertEqual(xcassetsEntries[2].parameters.count, 0)
+    } catch let e {
+      XCTFail("Error: \(e)")
+    }
+  }
+  // swiftlint:enable function_body_length
+
+  func testReadInvalidConfigThrows() {
+    let badConfigs = [
+      "config-missing-paths": "Missing entry for key strings.paths.",
+      "config-missing-template": "You must specify a template name (-t) or path (-p).\n\n" +
+      "To list all the available named templates, use 'swiftgen templates list'.",
+      "config-both-templates": "You need to choose EITHER a named template OR a template path. " +
+      "Found name 'template' and path 'template.swift'",
+      "config-missing-output": "Missing entry for key strings.output.",
+      "config-invalid-structure": "Wrong type for key strings.paths: expected Path or array of Paths, got Array<Any>.",
+      "config-invalid-template": "Wrong type for key strings.templateName: expected String, got Array<Any>.",
+      "config-invalid-output": "Wrong type for key strings.output: expected String, got Array<Any>."
+    ]
+    for (configFile, expectedError) in badConfigs {
+      do {
+        guard let path = Bundle(for: type(of: self)).path(forResource: configFile, ofType: "yml") else {
+          fatalError("Fixture not found")
+        }
+        _ = try Config(file: Path(path))
+        XCTFail("Trying to parse config file \(configFile) should have thrown " +
+          "error \(expectedError) but didn't throw at all")
+      } catch let e {
+        XCTAssertEqual(
+          String(describing: e), expectedError,
+          "Trying to parse config file \(configFile) should have thrown " +
+          "error \(expectedError) but threw \(e) instead."
+        )
+      }
+    }
+  }
+
+  // MARK: Linting
```

Comment:
Maybe move the config linting to another test file

Reply:
Good point, will do.